### ISSUE TITLE
bin/run fails: staticWhich: Could not find executable for "stork"

### ISSUE THREAD

**Issue by kukimik** [srid/emanote#359]
**Describe the bug**

When trying to build emanote (using bin/run or Ctrl+Shift+B in VSCode) i get:

```
 *  Executing task: bin/run 

+ export PORT=9010
+ PORT=9010
+ exec nix develop -c ghcid -c 'cabal repl exe:emanote' --warnings -T ':main -L ./docs run --port=9010'
warning: Using saved setting for 'extra-substituters = https://cache.garnix.io' from ~/.local/share/nix/trusted-settings.json.
warning: Using saved setting for 'extra-trusted-public-keys = cache.garnix.io:CTFPyKSLcx5RMJKfLo5EEPUObbA78b0YQ2DTCJXqr9g=' from ~/.local/share/nix/trusted-settings.json.
Loading cabal repl exe:emanote ...
Build profile: -w ghc-9.0.2 -O1
In order, the following will be built (use -v for more details):
 - emanote-0.7.12.3 (exe:emanote) (first run)
Preprocessing executable 'emanote' for emanote-0.7.12.3..
GHCi, version 9.0.2: https://www.haskell.org/ghc/  :? for help

<no location info>: warning: [-Wmissing-home-modules]
    These modules are needed for compilation but not listed in your .cabal file's other-modules: 
        Emanote
        Emanote.CLI
        Emanote.Model
        Emanote.Model.Calendar
        Emanote.Model.Graph
        Emanote.Model.Link.Rel
        Emanote.Model.Link.RelSpec
        Emanote.Model.Link.Resolve
        Emanote.Model.Meta
        Emanote.Model.Note
        Emanote.Model.Note.Filter
        Emanote.Model.Query
        Emanote.Model.QuerySpec
        Emanote.Model.SData
        Emanote.Model.StaticFile
        Emanote.Model.Stork
        Emanote.Model.Stork.Index
        Emanote.Model.Task
        Emanote.Model.Title
        Emanote.Model.Type
        Emanote.Pandoc.BuiltinFilters
        Emanote.Pandoc.Link
        Emanote.Pandoc.Markdown.Parser
        Emanote.Pandoc.Markdown.Syntax.HashTag
        Emanote.Pandoc.Markdown.Syntax.Highlight
        Emanote.Pandoc.Markdown.Syntax.WikiLink
        Emanote.Pandoc.Renderer
        Emanote.Pandoc.Renderer.Embed
        Emanote.Pandoc.Renderer.Query
        Emanote.Pandoc.Renderer.Url
        Emanote.Prelude
        Emanote.Route
        Emanote.Route.Ext
        Emanote.Route.ModelRoute
        Emanote.Route.R
        Emanote.Route.SiteRoute
        Emanote.Route.SiteRoute.Class
        Emanote.Route.SiteRoute.Type
        Emanote.Source.Dynamic
        Emanote.Source.Loc
        Emanote.Source.Patch
        Emanote.Source.Pattern
        Emanote.View.Common
        Emanote.View.Export
        Emanote.View.LiveServerFiles
        Emanote.View.TagIndex
        Emanote.View.TaskIndex
        Emanote.View.Template
        Paths_emanote
        Spec
[ 1 of 51] Compiling Emanote.Pandoc.Markdown.Syntax.HashTag ( src/Emanote/Pandoc/Markdown/Syntax/HashTag.hs, interpreted )
[ 2 of 51] Compiling Emanote.Pandoc.Markdown.Syntax.Highlight ( src/Emanote/Pandoc/Markdown/Syntax/Highlight.hs, interpreted )
[ 3 of 51] Compiling Emanote.Pandoc.Renderer ( src/Emanote/Pandoc/Renderer.hs, interpreted )
[ 4 of 51] Compiling Emanote.Prelude  ( src/Emanote/Prelude.hs, interpreted )
[ 5 of 51] Compiling Emanote.Model.Stork.Index ( src/Emanote/Model/Stork/Index.hs, interpreted )

src/Emanote/Model/Stork/Index.hs:58:13: error:
    • staticWhich: Could not find executable for "stork"
    • In the untyped splice: $(staticWhich "stork")
   |
58 | storkBin = $(staticWhich "stork")
   |             ^^^^^^^^^^^^^^^^^^^^^
Failed, four modules loaded.

src/Emanote/Model/Stork/Index.hs:58:13: error:
    • staticWhich: Could not find executable for "stork"
    • In the untyped splice: $(staticWhich "stork")
   |
58 | storkBin = $(staticWhich "stork")
   |     
```

This was introduced in [65dd75053674aaf4fc36c0ad689938c8d6158e2e.](https://github.com/EmaApps/emanote/commit/65dd75053674aaf4fc36c0ad689938c8d6158e2e).

**Desktop (please complete the following information):**
Linux 5.4.0-131-generic x86-64, Linux Mint 20.3

**Comment by kukimik** [comment]
`nix --version` shows `nix (Nix) 2.9.1`, so I think I'm not using the latest release 2.11.1. Maybe this is my problem? But I can't even  figure how to upgrade it. (Sorry, I have no idea about nix.) 

However, I have an excuse: [the readme](https://github.com/EmaApps/emanote#developing) says: `All but the final step need to be done only once.` and it does not mention upgrading.

**Comment by srid** [comment]
> This was introduced in [65dd75053674aaf4fc36c0ad689938c8d6158e2e.](https://github.com/EmaApps/emanote/commit/65dd75053674aaf4fc36c0ad689938c8d6158e2e).

When you add back `inherit (self'.packages) stork;` under `buildTools` does it work for you?

**Comment by srid** [comment]
Oh looks like this happens only on Linux. Not on my macOS M1 ...

**Comment by srid** [comment]
This is nothing to do with Nix version. The error comes from https://github.com/obsidiansystems/which/blob/develop/src/System/Which.hs#L21

But stork **should** be in PATH (do a `nix develop` and do `which stork` in it) such that `which` picks it up, especially after 944767a86152af3aab884e7c2ce5382c94162f33 - yet that error is still being thrown, but only on Linux. Strange.

**Comment by kukimik** [comment]
> When you add back inherit (self'.packages) stork; under buildTools does it work for you?

I also tried restoring it locally and it didn't work.

But now, when I pulled https://github.com/EmaApps/emanote/commit/944767a86152af3aab884e7c2ce5382c94162f33, it works. :open_mouth:  Don't know, maybe something somewhere is using the master branch at github instead of the state of the local repo?

Anyway, it works now for me, so if you're not into solving this mystery, the issue can probably be closed. Thanks!
